lib/robots/dor_repo/gis_assembly/generate_content_metadata.rb

```diff
@@ -14,86 +14,76 @@ def initialize
-
-                    mimetype = o.image? ? MIME::Types.type_for("xxx.#{FastImage.type(o.path)}").first.to_s : o.mimetype
-
-                    roletype = if mimetype == 'application/zip'
-                                 if o.path =~ /_(EPSG_\d+)/i # derivative
-                                   'derivative'
-                                 else
-                                   'master'
-                                 end
-                               elsif o.image?
-                                 if o.path =~ /_small.png$/
-                                   'derivative'
-                                 else
-                                   'master'
-                                 end
-                               end || nil
-
-                    o.file_attributes ||= {}
-                    [:publish, :shelve].each { |t| o.file_attributes[t] = 'yes' }
-                    o.file_attributes[:preserve] = if roletype == 'master'
-                                                     'yes'
-                                                   else
-                                                     'no'
-                                                   end
-
-                    xml.file o.file_attributes.merge(
-                      id: o.filename,
-                      mimetype: mimetype,
-                      size: o.filesize,
-                      role: roletype || 'master'
-                    ) do
-                      if resource_type == :object
-                        if roletype == 'master' && !geo_data_xml.nil?
-                          xml.geoData do
-                            xml.parent.add_child geo_data_xml
-                          end
-                          geo_data_xml = nil # only once
-                        elsif o.filename =~ /_EPSG_(\d+)\.zip/i
-                          xml.geoData srsName: "EPSG:#{Regexp.last_match(1)}"
-                        end
-                      end
-                      xml.checksum(o.sha1, type: 'sha1')
-                      xml.checksum(o.md5, type: 'md5')
-                      if o.image?
-                        wh = FastImage.size(o.path)
-                        xml.imageData width: wh[0], height: wh[1]
-                      end
-                    end
-                  end
-                  seq += 1
-                end
+        def create_content_metadata(cocina_model, druid, objects)
+          seq = 0
+          access = file_access(cocina_model)
+          objects.map do |k, v|
+            next if v.nil? || v.empty?
+
+            resource_type = case k
+                            when :Data
+                              Cocina::Models::FileSetType.object
+                            when :Preview
+                              Cocina::Models::FileSetType.preview
+                            else
+                              Cocina::Models::FileSetType.attachment
+                            end
+
+            files = v.map do |o|
```

Comment:
"o" is a terrible name - is it a filename?

Reply:
I don't think so, It appears to be some sort of object that has an `#image?` and `#path` method.

Reply:
assembly_objectfile would be a good name then, no?

Reply:
Okay, I went and fixed up all of the problems in this code.  I'm afraid it's harder to understand how this code relates to the old state now though.